contracts/account/extensions/AccountERC7579.sol

```diff
@@ -0,0 +1,370 @@
+
+        IERC7579Module(module).onUninstall(deInitData);
+        emit ModuleUninstalled(moduleTypeId, module);
+    }
+
+    /**
+     * @dev Installs a fallback handler for the given selector. Returns true if the handler was installed,
+     */
+    function _installFallback(address module, bytes4 selector) internal virtual returns (bool) {
+        if (_fallbacks[selector] != address(0)) return false;
+        _fallbacks[selector] = module;
+        return true;
+    }
+
+    /**
+     * @dev Uninstalls a fallback handler for the given selector. Returns true if the handler was uninstalled,
+     */
+    function _uninstallFallback(address module, bytes4 selector) internal virtual returns (bool) {
+        address handler = _fallbackHandler(selector);
+        if (handler == address(0) || handler != module) return false;
+        delete _fallbacks[selector];
+        return true;
+    }
+
+    /// @dev Returns the fallback handler for the given selector. Returns `address(0)` if not installed.
+    function _fallbackHandler(bytes4 selector) internal view virtual returns (address) {
+        return _fallbacks[selector];
+    }
+
+    /// @dev Extracts the nonce validator from the user operation.
+    ///
+    /// To construct a nonce key, set nonce as follows:
+    ///
+    /// ```
+    /// <module address (20 bytes)> | <key (4 bytes)> | <nonce (8 bytes)>
+    /// ```
+    function _extractUserOpValidator(PackedUserOperation calldata userOp) internal pure virtual returns (address) {
+        return address(bytes32(userOp.nonce).extract_32_20(0));
+    }
+
+    /// @dev Extracts the signature validator from the signature.
+    ///
+    /// To construct a signature, set the first 20 bytes as the module address and the remaining bytes as the
+    /// signature data:
+    ///
+    /// ```
+    /// <module address (20 bytes)> | <signature data>
+    /// ```
+    function _extractSignatureValidator(bytes calldata signature) internal pure virtual returns (address) {
+        return address(bytes20(signature[0:20]));
+    }
+
+    /**
+     * @dev Fallback function that delegates the call to the installed handler for the given selector.
+     *
+     * Reverts with {ERC7579MissingFallbackHandler} if the handler is not installed.
+     *
+     * Calls the handler with the original `msg.sender` appended at the end of the calldata following
+     * the ERC-2771 format.
+     */
+    function _fallback() internal virtual {
+        address handler = _fallbackHandler(msg.sig);
+        require(handler != address(0), ERC7579MissingFallbackHandler(msg.sig));
+
+        // From https://eips.ethereum.org/EIPS/eip-7579#fallback[ERC-7579 specifications]:
+        // - MUST utilize ERC-2771 to add the original msg.sender to the calldata sent to the fallback handler
+        // - MUST use call to invoke the fallback handler
+        (bool success, bytes memory returndata) = handler.call{value: msg.value}(
+            abi.encodePacked(msg.data, msg.sender)
```

Comment:
shoud we use Context here ?

I know context is going to go away, and I understand that its unlikelly this will be used with an ERC2771Context ... but it could happen and I wouldn't create inconsistencies in our usage of Context in 5.x

```suggestion
            abi.encodePacked(_msgData(), _msgSender())
```